README.md

```diff
@@ -0,0 +1,24 @@
+# GOV.UK Repository Mirrorer
+
+Ensures all repositories tagged with "govuk" in the alphagov organisation in Github.com are mirrored to Gitlab.com.
+
+We use Gitlab.com so we can still deploy in the event that Github.com is down. We also use the private repositories at Gitlab in order to fix any security vulnerabilities without disclosing them.
+
+This script will ensure a project exists in the `govuk` namespace in Gitlab.com for every `govuk` Github repository. It will then synchronise all branches and tags, including removing branches that have been deleted in Github. It is designed to be run periodically in Jenkins.
+
+**Warning:** If you need to work on a branch in private on Gitlab.com, this script must be prevented from running. If it runs and the branch you're working on does not exist in Github, it will be removed from Gitlab.
```

Comment:
I wonder if there's a way to "lock" an individual GitLab project such that the mirroring process can say "skip it," rather than stopping mirroring for _all_ projects.

It's possible to add "project tags," similar to repo labels in GitHub, so a project in GitLab via the settings (Settings → General → General project settings → Tags), but these [aren't available in the API](https://gitlab.com/gitlab-org/gitlab-ce/issues/2978) at the moment. Maybe something to think about at a later date.

Reply:
Probably easiest to just untag the repo as "govuk" temporarily. Will add to the readme.